Re-scaffold and BLUI-fy

daileytj opened the pull request:
<!-- If this pull request fixes an Issue, link it below. If not, you can remove the line below -->
Fixes PXBLUE-2516.

<!-- Include a bulleted list summarizing the main changes you have made in this PR -->
#### Changes proposed in this Pull Request:
- Rescaffold this repo using RNCC instead of Expo
- Swap out PX Blue references for Brightlayer UI

There is a known issue with the CollapsibleHeader example: https://github.com/pxblue/react-native-component-library/issues/228

huayunh commented:
So tired of RN... I'd rather just see screenshots

> yarn run v1.22.10
$ react-native run-ios
error React Native CLI uses autolinking for native dependencies, but the following modules are linked manually: 
>  - @brightlayer-ui/react-native-vector-icons (to unlink run: "react-native unlink @brightlayer-ui/react-native-vector-icons")
>  - react-native-vector-icons (to unlink run: "react-native unlink react-native-vector-icons")
> 
> This is likely happening when upgrading React Native from below 0.60 to 0.60 or above. Going forward, you can unlink this dependency via "react-native unlink <dependency>" and it will be included in your app automatically. If a library isn't compatible with autolinking, disregard this message and notify the library maintainers.

> error Failed to build iOS project. We ran "xcodebuild" command but it exited with error code 65. To debug build logs further, consider building your app with Xcode.app, by opening ReactNativeDesignPatterns.xcworkspace.
> Command line invocation:
>    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild -workspace ReactNativeDesignPatterns.xcworkspace -configuration Debug -scheme ReactNativeDesignPatterns -destination id=23D87531-AB4A-45D6-B821-675E8CF1E55B

daileytj commented:
Fair enough... I'll post some screenshots or a video for you on Monday. I'll still need some regression testing done with functionality though.

jeffvg commented:
everything looks good on ios/android and all pxb refs are removed from what I can see. Since we are scaffolded need to update the root read me for running locally on iOS

daileytj commented:
@huayunh Here's a screen recording going through each of the screens. I noticed the drag handle icon size is off on the "sortable list" example, so I'll take care of that. Other than that this should be good to review (the UI anyway... still have to fix tests :)). 
https://user-images.githubusercontent.com/13989985/142908570-8e299db9-d2d2-462b-b1fb-38361a8f2b16.mov

daileytj commented:
These tests are driving me crazy and I have no idea why certain tests are failing. So, We will need to go back in and re-add the tests I commented out for the action list, multi-select list, and status list. I'll write up a story to fix these unless someone knows how to fix these.

huayunh commented:
Everything under forms&validation, since this is the only one we implemented new patterns for: 

(Feel free to log bugs for these)

# 1

On the Password validation screen, the button would need to be above the "go back to home screen" line (don't know what it is called)

![image](https://user-images.githubusercontent.com/8997218/142957468-631b502e-540e-4bac-924b-5db4d4152eb6.png)

# 2

![image](https://user-images.githubusercontent.com/8997218/142957612-23e76392-e618-4cb0-8f95-7a0c5fb06ed3.png)

The visibility toggle button shall be gray500

Something else I noticed besides the dainty sorting handle you pointed out:

# 3
Sortable list, I don't think those numbers are rendered in Open Sans

![image](https://user-images.githubusercontent.com/8997218/142957921-5be8e65c-db61-41c6-9cf8-ef1abdb74e2b.png)

emclaug2 commented:
Another issue for the pile: 
![image](https://user-images.githubusercontent.com/6538289/143028877-8f3ab4e8-29c8-45c1-bdd6-4204d3146ff1.png)

This button doesn't fit on the screen anymore due to the name change.

emclaug2 commented:
Piggybacking on what @huayunh pointed out, on android the Password Validation button appear off screen: 

![image](https://user-images.githubusercontent.com/6538289/143030096-f9b679d4-52f0-4eba-b345-8a5e5df436af.png)


These not a part of the BLUI migration work but'll need some new bugs.

jeffvg commented:
> Piggybacking on what @huayunh pointed out, on android the Password Validation button appear off screen:
> 
> ![image](https://user-images.githubusercontent.com/6538289/143030096-f9b679d4-52f0-4eba-b345-8a5e5df436af.png)
> 
> These not a part of the BLUI migration work but'll need some new issues made.

just curious how you get to this state? Is the device keyboard enabled or are you dismissing the keyboard?

emclaug2 commented:
> just curious how you get to this state? Is the device keyboard enabled or are you dismissing the keyboard?

I'm not sure, I thought I had just dismissed the keyboard like normal.

daileytj commented:
Fixed typography...
![Simulator Screen Shot - iPhone 12 - 2021-11-23 at 09 20 48](https://user-images.githubusercontent.com/13989985/143042508-a4518730-735a-4c96-b282-e87a42a35878.png)
Fixed icon color...
![Simulator Screen Shot - iPhone 12 - 2021-11-23 at 09 20 44](https://user-images.githubusercontent.com/13989985/143042511-ad81135b-f29c-4564-83e2-d7618d796e7b.png)
Fixed button issue... hopefully this is an ok solution...
![Simulator Screen Shot - iPhone 12 - 2021-11-23 at 09 20 33](https://user-images.githubusercontent.com/13989985/143042512-518c6ec9-8d29-417e-934d-28a7d71ae23a.png)

Going to write a bug for the submit button thing... 

@jeffvg @emclaug2 @huayunh Should be good for a second look 👀

jeffvg commented:
weird, I'll have to look in dev branch. Android page not scrollable to view submit

![image](https://user-images.githubusercontent.com/37150565/143097741-ac737a71-0434-4721-82cc-cfbb696f737e.png)

jeffvg commented on README.md:
````diff
@@ -1,30 +1,30 @@
+git clone https://github.com/brightlayer-ui/react-native-design-patterns.git
 cd react-native-design-patterns
 ```
 
````
yarn && yarn ios should be updated to cd ios "pod install" cd... yarn ios

huayunh commented:
If you wonder how far that button shall be, here is a screenshot from an Eaton project: 
![image](https://user-images.githubusercontent.com/8997218/143119362-4919a604-3144-443f-87d5-b0cc2cfe358f.png)
